modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/ExtensionsClientTestCase.java

```diff
@@ -0,0 +1,257 @@
+        .addParameter("type", FIRE_WEAPON)
+        .addParameter("attributesOfWeapon", attributes)
+        .build();
+    Result<String, Attributes> result = client.execute(HEISENBERG_EXT_NAME, "killWithWeapon", params);
+    assertThat(result.getOutput(), is("Killed with: You have been killed with Ricin , Type FIRE_WEAPON and attribute brand"));
+  }
+
+  @Test
+  @Description("Executes an operation that has a parameter group using the client and checks the output")
+  public void executeOperationWithParameterGroup() throws MuleException {
+    OperationParameters params = builder().configName(HEISENBERG_CONFIG)
+        .addParameter("greeting", "jeje")
+        .addParameter("age", 23)
+        .addParameter("name", "Juani")
+        .addParameter("knownAddresses", emptyList())
+        .build();
+    Result<String, Attributes> result = client.execute(HEISENBERG_EXT_NAME, "alias", params);
+    assertThat(result.getOutput(), is(ALIAS_OUTPUT));
+  }
+
+  @Test
+  @Description("Executes an operation that has a parameter with default value using the client and checks the output")
+  public void executeOperationWithDefaultValueParameter() throws MuleException {
+    OperationParameters params = builder().configName("apple").build();
+    Result<Collection<?>, Attributes> result = client.execute(VEGAN, "tryToEatThisListOfMaps", params);
+    assertThat(result.getOutput(), instanceOf(List.class));
+    assertThat(result.getOutput(), hasSize(0));
+  }
+
+  @Test
+  @Description("Executes an operation that has a @NullSafe annotated parameter using the client and checks the output")
+  public void executeOperationWithNullSafeParameter() throws MuleException {
+    OperationParameters params = builder().configName("banana").build();
+    Result<VeganPolicy, Attributes> result = client.execute(VEGAN, "applyPolicy", params);
+    assertThat(result.getOutput().getMeetAllowed(), is(false));
+    assertThat(result.getOutput().getIngredients().getSaltMiligrams(), is(0));
+  }
+
+  @Test
+  @Description("Executes an operation that returns attributes metadata using the client and checks the output and the attributes")
+  public void executeOperationThatReturnsAttributes() throws MuleException {
+    OperationParameters params = builder().configName(HEISENBERG_CONFIG).build();
+    Result<String, IntegerAttributes> result = client.execute(HEISENBERG_EXT_NAME, "getEnemy", params);
+    assertThat(result.getOutput(), is("Gustavo Fring"));
+    assertThat(result.getAttributes().isPresent(), is(true));
+    assertThat(result.getAttributes().get().getValue(), is(0));
+  }
+
+  @Test
+  @Description("Executes a void operation using the client")
+  public void executeVoidOperation() throws MuleException {
+    OperationParameters params = builder().configName(HEISENBERG_CONFIG).build();
+    Result<Object, Attributes> result = client.execute(HEISENBERG_EXT_NAME, "die", params);
+    assertThat(result.getOutput(), is(nullValue()));
```

Comment:
for now yes. The alternative would be changing the value for an Optional but not sure if that's a good idea to introduce right now